apps/extension/src/core/media/model.ts

```diff
@@ -0,0 +1,239 @@
+      getRoutePath: config.getRoutePath || ((request: RequestData) => null),
+      overrideModelParam:
+        config.overrideModelParam || ((request: RequestData) => request.model),
+      cacheGet: config.cacheGet || (() => Promise.resolve(undefined)),
+      cacheSet: config.cacheSet || (() => Promise.resolve(undefined))
+    }
+    return opts
+  }
+
+  log(...args: unknown[]): void {
+    if (this.config.debug) {
+      console.log(`[MEDIA_MODEL ${this.config.identifier}]: `, ...args)
+    }
+  }
+
+  error(...args: unknown[]): void {
+    if (this.config.debug) {
+      console.error(`[MEDIA_MODEL ${this.config.identifier}]: `, ...args)
+    }
+  }
+
+  getRequestIdentifierData(
+    requestPrompt: RequestPrompt,
+    opts: Required<RequestOptions>
+  ): RequestData {
+    const ret = {
+      ...requestPrompt,
+      model: opts.model,
+      identifier: this.config.identifier,
+      num_generations: opts.num_generations,
+      num_inference_steps: opts.num_inference_steps,
+      mime_type: opts.mime_type,
+      baseUrl: opts.baseUrl
+    }
+    return {
+      ...ret,
+      model: this.config.overrideModelParam(ret)
+    }
+  }
+
+  async generate(
+    requestPrompt: RequestPrompt,
+    requestOpts: RequestOptions = {}
+  ): Promise<Result<MediaOutput[], ErrorCode>> {
+    const {
+      transformForRequest,
+      getPath,
+      transformResponse
+    } = this.config
+    const opts: Required<RequestOptions> = {
+      ...this.defaultOptions,
+      ...definedValues(requestOpts)
+    }
+    const request = this.getRequestIdentifierData(requestPrompt, opts)
+    const id = objectHash(request)
+    const promptSnippet = JSON.stringify(requestPrompt).slice(0, 100)
+    const payload = transformForRequest(request, opts)
+    this.log(`COMPLETING id ${id}: ${promptSnippet}...`, {
+      modelId: request.model,
+    })
+    let responseData: Record<string, any>
+    try {
+      const response = await this.api.post(getPath(request), payload, {
+        baseURL: opts.baseUrl,
+        timeout: opts.timeout,
+        headers: this._getRequestHeaders(opts)
+      })
+      responseData = response.data
+    } catch (err: unknown) {
+      console.log(err)
+      return this._handleModelAPIError(err)
+    }
+
+    this.log("RESPONSE for id " + id)
+    // return ok([responseData.url]);
+    const result = transformResponse(responseData)
+    if (!result[0]) {
+      const e = new Error(
+        `Returned an empty result: ${JSON.stringify(responseData)}`
+      )
+      this.error(e)
+      throw e
+    }
+    return ok(result)
+  }
+
+
+
+  protected _getRequestHeaders(opts: Required<RequestOptions>) {
+    const { authPrefix } = this.config
+    return {
+      Authorization: opts.apiKey ? `${authPrefix}${opts.apiKey}` : undefined,
+      "X-API-KEY": opts.apiKey || undefined,
+      "HTTP-Referer": opts.origin
+    }
+  }
+
+  private _handleModelAPIError(error: unknown): Err<ErrorCode> {
```

Comment:
nit: i think we should write a little "TODO" comment above methods that we know are duplicated from the other model.ts class, to hint that we're just duplicating things temporarily

Reply:
added